packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1375,6 +1376,61 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  /**
+   * Updates the editable parameters of a transaction.
+   *
+   * @param txId - The ID of the transaction to update.
+   * @param params - The editable parameters to update.
+   * @param params.data - Data to pass with the transaction.
+   * @param params.gas - Maximum number of units of gas to use for the transaction.
+   * @param params.gasPrice - Price per gas for legacy transactions.
+   * @param params.from - Address to send the transaction from.
+   * @param params.to - Address to send the transaction to.
+   * @param params.value - Value associated with the transaction.
+   * @returns The updated transaction metadata.
+   */
+  async updateEditableParams(
+    txId: string,
+    { data, gas, gasPrice, from, to, value }: EditableTransactionParams,
+  ) {
+    const transactionMeta = this.getTransaction(txId);
+    if (!transactionMeta) {
+      throw new Error(
+        `Cannot update editable params as no transaction metadata found`,
+      );
+    }
+
+    validateIfTransactionUnapproved(transactionMeta, 'updateEditableParams');
+
+    const editableParams: Partial<TransactionMeta> = {
+      txParams: {
+        data,
+        from: from ?? transactionMeta.txParams.from,
+        to,
+        value,
+        gas,
+        gasPrice,
+      },
+    };
+    editableParams.txParams = pickBy(
+      editableParams.txParams,
+      (prop) => prop !== undefined,
```

Comment:
Do we need this argument, I think the default is to remove any `undefined` values?

Reply:
yes, I just removed it for simplicity https://github.com/MetaMask/core/pull/2056/commits/87b3ecbb7b2e039a3271ca9385cd176c1192aa18.